### ISSUE TITLE
Evaluate accuracy of steppers with interpolation

### ISSUE THREAD

**Issue by jonapost** [jonapost/field_propagation#3]
Test the accuracy of steppers with interpolation. 

Plot deviation from exact solution (and estimated error) for existing step sizes (pi/4, pi/3, pi/2) and large ones: 3pi/4, pi, 5 pi/4, 3pi/2, 2_pi, 2.5_pi

**Comment by somnergy** [comment]
The first version of this test has been implemented. It does steps in theta (rotation) as specified by command line values. An update is underway for this test, the idea is to simply create one (or more) class to handle the variables, output and the functions.
The <em> G4ExactHelixStepper's </em> solution have been treated as the exact solution in this case (Uniform Magnetic Field) and deviation from it has been treated as error.
The plot of <em> maxErr <em> vs </em> step_size </em> can be made from the results. Can you please mention what other plots you want to see ? Can you also clarify what are you referring to as "estimated error" ?

**Comment by somnergy** [comment]
I <a href="https://github.com/jonapost/field_propagation/commits/interpol-v1">commit</a>ted the <a href="https://github.com/jonapost/field_propagation/commit/f027de261301a16b0f9b5011d5ef60a54f558ddd" > code </a> to the branch  <a href="https://github.com/jonapost/field_propagation/tree/interpol-v1"> interpol-v1 </a>

**Comment by jonapost** [comment]
Thanks. 
Estimated error = the error estimate obtained from the Stepper (argument yerr) 

Please adapt / normalise it/them as follows into a few plots:
- one plot for the error for all positions,  || Dx || - the magnitude of the 3-vector of the position part of yerr - to be compared with the deviation;
- one plot for the error in momentum,  || Dp || / ||p||- the magnitude of the 3-vector of momentum in yerr. Please compare on the same plot with the difference between the momentum estimated and the correct/perfect momentum (from the helix in the case of the uniform field, as you said).

( Only quantities with the same units should be plotted together - i.e. the momentum error should be plotted on a separate plot, together with the momentum deviation )

**Comment by somnergy** [comment]
Thanks for the clarifications. Your requests are clear to me. I want to inform you that the code above only tests the accuracy and gives the output and/or maxErr per coordinate. Also estimated error has not been included (for output) as it is not available at the non-mesh points. Did you mean to exclude the accuracy of interpolants and evaluate the accuracy at the mesh points only ? (Which had been done earlier, results stored safely)

**Comment by jonapost** [comment]
You are correct that there is no estimate of the error of the interpolant.  I will need to move those requests to a different issue - regarding the error of the endpoints.
We need only the _deviation_ of the interpolant from the 'exact' values - for position (x) as a vector (initially it is ok to show it by component, but I also want the magnitude of the difference as a vector) and for momentum p as a vector (by component at first, and also the mag of the difference.)

**Comment by jonapost** [comment]
Your observation blows a hole in my (erroneous) request.  Well caught.  I was mixed up. In a first iteration please forget the request for estimated error.  

But I think that it would likely be quite interesting to compare the deviation with the estimated error for a step that has as its step size the same length.  We can figure out whether it is interesting enough to do now or not!
